### PR TITLE
ARROW-8591: [Rust] Reverse lookup for a key in DictionaryArray

### PR DESCRIPTION
**Pull request by vertexclique**
This PR enables reverse lookup for already built dict.

### CODE REVIEW

File: rust/arrow/src/array/array.rs [hunk 1]
```diff
@@ -1786,38 +1786,34 @@ impl From<(Vec<(Field, ArrayRef)>, Buffer, usize)> for StructArray {
...
     data: ArrayDataRef,
 
-    // Pointer to the key values.
+    /// Pointer to the key values.
     raw_values: RawPtrBox<K::Native>,
 
-    // Array of any values.
+    /// Array of any values.
     values: ArrayRef,
-
```
Comment (nevi-me) [inline rust/arrow/src/array/array.rs:1818]: `is_ordered` is needed for IPC purposes, so we shouldn't remove it for compatibility purposes

Comment (vertexclique) [reply rust/arrow/src/array/array.rs:1818]: Brought it back.

**Review by paddyhoran — APPROVED** [review]
LGTM

**Comment by vertexclique** [comment]
Do you have any ETA for merging this PR? Currently, I am using my local copy but I want to update to the git dep.

**Comment by paddyhoran** [comment]
Was just giving other committers a chance to review.  If there is no update, I'll merge later today.

**Pull request closed without merging**